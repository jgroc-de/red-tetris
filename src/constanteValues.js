export const numberOfLines = 20
export const numberOfColumns = 10
export const filledSquared = '-'
export const filledTetriminosSquared = 'x'
export const emptySquared = ' '
export const rotationCases = 4
export const LShape = [
    [
        emptySquared + emptySquared + filledTetriminosSquared,
        filledTetriminosSquared + filledTetriminosSquared + filledTetriminosSquared,
        emptySquared + emptySquared + emptySquared,
    ],
    [
        emptySquared + filledTetriminosSquared + emptySquared,
        emptySquared + filledTetriminosSquared + emptySquared,
        emptySquared + filledTetriminosSquared + filledTetriminosSquared,
    ],
    [
        emptySquared + emptySquared + emptySquared,
        filledTetriminosSquared + filledTetriminosSquared + filledTetriminosSquared,
        filledTetriminosSquared + emptySquared + emptySquared,
    ],
    [
        filledTetriminosSquared + filledTetriminosSquared + emptySquared,
        emptySquared + filledTetriminosSquared + emptySquared,
        emptySquared + filledTetriminosSquared + emptySquared,
    ]
]
export const JShape = [
    [
        filledTetriminosSquared + emptySquared + emptySquared,
        filledTetriminosSquared + filledTetriminosSquared + filledTetriminosSquared,
        emptySquared + emptySquared + emptySquared,
    ],
    [
        emptySquared + filledTetriminosSquared + filledTetriminosSquared,
        emptySquared + filledTetriminosSquared + emptySquared,
        emptySquared + filledTetriminosSquared + emptySquared,
    ],
    [
        emptySquared + emptySquared + emptySquared,
        filledTetriminosSquared + filledTetriminosSquared + filledTetriminosSquared,
        emptySquared + emptySquared + filledTetriminosSquared,
    ],
    [
        emptySquared + filledTetriminosSquared + emptySquared,
        emptySquared + filledTetriminosSquared + emptySquared,
        filledTetriminosSquared + filledTetriminosSquared + emptySquared,
    ]
]
export const SShape = [
    [
        emptySquared + filledTetriminosSquared + filledTetriminosSquared,
        filledTetriminosSquared + filledTetriminosSquared + emptySquared,
        emptySquared + emptySquared + emptySquared,
    ],
    [
        emptySquared + filledTetriminosSquared + emptySquared,
        emptySquared + filledTetriminosSquared + filledTetriminosSquared,
        emptySquared + emptySquared + filledTetriminosSquared,
    ],
    [
        emptySquared + emptySquared + emptySquared,
        emptySquared + filledTetriminosSquared + filledTetriminosSquared,
        filledTetriminosSquared + filledTetriminosSquared + emptySquared,
    ],
    [
        filledTetriminosSquared + emptySquared + emptySquared,
        filledTetriminosSquared + filledTetriminosSquared + emptySquared,
        emptySquared + filledTetriminosSquared + emptySquared,
    ]
]
export const OShape = [
    [
        emptySquared + filledTetriminosSquared + filledTetriminosSquared + emptySquared,
        emptySquared + filledTetriminosSquared + filledTetriminosSquared + emptySquared,
        emptySquared + emptySquared + emptySquared + emptySquared,
        emptySquared + emptySquared + emptySquared + emptySquared,
    ],
    [
        emptySquared + filledTetriminosSquared + filledTetriminosSquared + emptySquared,
        emptySquared + filledTetriminosSquared + filledTetriminosSquared + emptySquared,
        emptySquared + emptySquared + emptySquared + emptySquared,
        emptySquared + emptySquared + emptySquared + emptySquared,
    ],
    [
        emptySquared + filledTetriminosSquared + filledTetriminosSquared + emptySquared,
        emptySquared + filledTetriminosSquared + filledTetriminosSquared + emptySquared,
        emptySquared + emptySquared + emptySquared + emptySquared,
        emptySquared + emptySquared + emptySquared + emptySquared,
    ],
    [
        emptySquared + filledTetriminosSquared + filledTetriminosSquared + emptySquared,
        emptySquared + filledTetriminosSquared + filledTetriminosSquared + emptySquared,
        emptySquared + emptySquared + emptySquared + emptySquared,
        emptySquared + emptySquared + emptySquared + emptySquared,
    ],
]
export const TShape = [
    [
        emptySquared + filledTetriminosSquared + emptySquared,
        filledTetriminosSquared + filledTetriminosSquared + filledTetriminosSquared,
        emptySquared + emptySquared + emptySquared,
    ],
    [
        emptySquared + filledTetriminosSquared + emptySquared,
        emptySquared + filledTetriminosSquared + filledTetriminosSquared,
        emptySquared + filledTetriminosSquared + emptySquared,
    ],
    [
        emptySquared + emptySquared + emptySquared,
        filledTetriminosSquared + filledTetriminosSquared + filledTetriminosSquared,
        emptySquared + filledTetriminosSquared + emptySquared,
    ],
    [
        emptySquared + filledTetriminosSquared + emptySquared,
        filledTetriminosSquared + filledTetriminosSquared + emptySquared,
        emptySquared + filledTetriminosSquared + emptySquared,
    ]
]
export const ZShape = [
    [
        filledTetriminosSquared + filledTetriminosSquared + emptySquared,
        emptySquared + filledTetriminosSquared + filledTetriminosSquared,
        emptySquared + emptySquared + emptySquared,
    ],
    [
        emptySquared + emptySquared + filledTetriminosSquared,
        emptySquared + filledTetriminosSquared + filledTetriminosSquared,
        emptySquared + filledTetriminosSquared + emptySquared,
    ],
    [
        emptySquared + emptySquared + emptySquared,
        filledTetriminosSquared + filledTetriminosSquared + emptySquared,
        emptySquared + filledTetriminosSquared + filledTetriminosSquared,
    ],
    [
        emptySquared + filledTetriminosSquared + emptySquared,
        filledTetriminosSquared + filledTetriminosSquared + emptySquared,
        filledTetriminosSquared + emptySquared + emptySquared,
    ],

]
export const IShape = [
    [
        emptySquared + emptySquared + emptySquared + emptySquared,
        filledTetriminosSquared + filledTetriminosSquared + filledTetriminosSquared + filledTetriminosSquared,
        emptySquared + emptySquared + emptySquared + emptySquared,
        emptySquared + emptySquared + emptySquared + emptySquared,
    ],
    [
        emptySquared + emptySquared + filledTetriminosSquared + emptySquared,
        emptySquared + emptySquared + filledTetriminosSquared + emptySquared,
        emptySquared + emptySquared + filledTetriminosSquared + emptySquared,
        emptySquared + emptySquared + filledTetriminosSquared + emptySquared,
    ],
    [
        emptySquared + emptySquared + emptySquared + emptySquared,
        emptySquared + emptySquared + emptySquared + emptySquared,
        filledTetriminosSquared + filledTetriminosSquared + filledTetriminosSquared + filledTetriminosSquared,
        emptySquared + emptySquared + emptySquared + emptySquared,
    ],
    [
        emptySquared + filledTetriminosSquared + emptySquared + emptySquared,
        emptySquared + filledTetriminosSquared + emptySquared + emptySquared,
        emptySquared + filledTetriminosSquared + emptySquared + emptySquared,
        emptySquared + filledTetriminosSquared + emptySquared + emptySquared,
    ],
]
