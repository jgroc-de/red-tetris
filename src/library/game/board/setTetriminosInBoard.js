import insertInBoard from './insertInBoard'

function setTetriminosInBoard(fullBoard, shape, position, numberOfColumns, emptySquared) {
    let updatedBoard = fullBoard

    for (let y = 0; y < shape.length; y++) {
        for (let x = 0; x < shape[0].length; x++) {
            let posY = (position.y + y) * numberOfColumns
            let posX = position.x + x
            if (shape[y][x] !== emptySquared) {
                updatedBoard = insertInBoard(updatedBoard, posX + posY, shape[y][x])
            }
        }
    }
    return updatedBoard
}

export default setTetriminosInBoard
