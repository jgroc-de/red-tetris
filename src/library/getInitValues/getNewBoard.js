function getNewBoard(length, emptySquared) {
    return emptySquared.repeat(length)
}

export default getNewBoard
