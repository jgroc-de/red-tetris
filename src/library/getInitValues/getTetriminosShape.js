import { IShape, JShape, LShape, OShape, SShape, TShape, ZShape } from "../../constanteValues"

function getTetriminosShape(shape) {
    switch (shape) {
        case 'I':
            return IShape
        case 'J':
            return JShape
        case 'L':
            return LShape
        case 'O':
            return OShape
        case 'S':
            return SShape
        case 'T':
            return TShape
        case 'Z':
            return ZShape
        default:
            return OShape
    }
}

export default getTetriminosShape