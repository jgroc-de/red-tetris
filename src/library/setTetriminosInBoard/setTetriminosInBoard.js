import {numberOfColumns, emptySquared} from '../../constanteValues'
import insertInBoard from './insertInBoard'

function setTetriminosInBoard(fullBoard, shape, position, token) {
    let updatedBoard = fullBoard

    for (let y = 0; y < shape.length; y++) {
        for (let x = 0; x < shape[0].length; x++) {
            let posY = (position.y + y) * numberOfColumns
            let posX = position.x + x
            if (shape[y][x] !== emptySquared) {
                updatedBoard = insertInBoard(updatedBoard, posX + posY, token)
            }
        }
    }
    return updatedBoard
}

export default setTetriminosInBoard
