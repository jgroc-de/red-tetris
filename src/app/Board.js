import Row from './Row'
import setTetriminosInBoard from '../library/setTetriminosInBoard/setTetriminosInBoard'
import {numberOfLines, numberOfColumns, filledTetriminosSquared} from '../constanteValues'

function Board(props) {
    const fullBoard = setTetriminosInBoard(props.board, props.shape, props.position, filledTetriminosSquared)
    const rows = []
    for (let i = 0; i < numberOfLines; i++) {
        rows.push(<Row
            key={i}
            number={i}
            squares={fullBoard.slice(i * numberOfColumns, i * numberOfColumns + numberOfColumns)}
        />)
    }

    return (
        <div id='board'>
            {rows}
        </div>
    )
}

export default Board
