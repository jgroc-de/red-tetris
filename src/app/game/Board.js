import Row from './Row'
import setTetriminosInBoard from '../../library/setTetriminosInBoard/setTetriminosInBoard'
import {numberOfLines, numberOfColumns, emptySquared} from '../../constanteValues'
import checkIfTetriminosCanMove from '../../library/checkIfTetriminosCanMove/checkIfTetriminosCanMove'
import getShapeType from '../../library/checkIfTetriminosCanMove/getShapeType'

function getGhostShape(shape) {
    let ghostShape = []
    let type = getShapeType(shape)
    for (let line of shape) {
        ghostShape.push(line.slice().replaceAll(type, 'g'))
    }

    return ghostShape
}

function getFullBoard(props, numberOfColumns, numberOfLines, emptySquared) {
    let position = {x: props.position.x, y: props.position.y + 1}
    while (checkIfTetriminosCanMove(props.board, props.shape, position, numberOfLines, numberOfColumns, emptySquared)) {
        ++position.y
    }
    position.y -= 1
    let fullBoard = setTetriminosInBoard(props.board, getGhostShape(props.shape), position, numberOfColumns, emptySquared)
    fullBoard = setTetriminosInBoard(fullBoard, props.shape, props.position, numberOfColumns, emptySquared)
    return fullBoard
}

function Board(props) {
    const fullBoard = getFullBoard(props, numberOfColumns, numberOfLines, emptySquared)
    const rows = []
    for (let i = 0; i < numberOfLines; i++) {
        rows.push(<Row
            key={i}
            number={i}
            width={numberOfColumns}
            squares={fullBoard.slice(i * numberOfColumns, i * numberOfColumns + numberOfColumns)}
        />)
    }

    return (
        <div id='board'>
            {rows}
        </div>
    )
}

export default Board
