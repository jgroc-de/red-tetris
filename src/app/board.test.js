import React from 'react'
import renderer from 'react-test-renderer'
import Board from './Board'
import  {numberOfColumns, numberOfLines, filledTetriminosSquared, emptySquared} from '../constanteValues'
import setTetriminosInBoard from '../library/setTetriminosInBoard/setTetriminosInBoard'
import getNewBoard from '../library/getInitValues/getNewBoard'
import getTetriminosShape from '../library/getInitValues/getTetriminosShape'

test('board with 1 tetriminos', () => {
    let board = getNewBoard(numberOfLines * numberOfColumns, emptySquared)
    let tetriminosShape = getTetriminosShape('O')[0]
    let tetriminosPosition = { x: 4, y: 3 }
    board = setTetriminosInBoard(board, tetriminosShape, tetriminosPosition, filledTetriminosSquared)
    const component = renderer.create(<Board board={board} shape={tetriminosShape} position={tetriminosPosition} />)
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
})