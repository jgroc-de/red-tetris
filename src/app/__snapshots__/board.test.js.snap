// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`board with 1 tetriminos 1`] = `
<div
  id="board"
>
  <div
    className="is-primary"
  >
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
  </div>
  <div
    className="is-primary"
  >
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
  </div>
  <div
    className="is-primary"
  >
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
  </div>
  <div
    className="is-primary"
  >
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square colored"
    />
    <div
      className="square colored"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
  </div>
  <div
    className="is-primary"
  >
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square colored"
    />
    <div
      className="square colored"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
  </div>
  <div
    className="is-primary"
  >
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
  </div>
  <div
    className="is-primary"
  >
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
  </div>
  <div
    className="is-primary"
  >
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
  </div>
  <div
    className="is-primary"
  >
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
  </div>
  <div
    className="is-primary"
  >
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
  </div>
  <div
    className="is-primary"
  >
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
  </div>
  <div
    className="is-primary"
  >
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
  </div>
  <div
    className="is-primary"
  >
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
  </div>
  <div
    className="is-primary"
  >
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
  </div>
  <div
    className="is-primary"
  >
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
  </div>
  <div
    className="is-primary"
  >
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
  </div>
  <div
    className="is-primary"
  >
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
  </div>
  <div
    className="is-primary"
  >
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
  </div>
  <div
    className="is-primary"
  >
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
  </div>
  <div
    className="is-primary"
  >
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
    <div
      className="square"
    />
  </div>
</div>
`;
